### ISSUE TITLE
Accessing Raw POST Body?

### ISSUE THREAD

**Issue by nick-andren** [kemalcr/kemal#321]
Hello,

I'm trying to POST a JSON blob to a Kemal server, but I'm not sure how to read it. Using variables like in a form will get them automatically parsed, but something like this becomes more difficult:
```
curl -X POST -d '{"testing":[1,2,3]}' 0.0.0.0:3000/testing
```
I have a route set up like this:
```
post "/testing" do |env|
  puts "#{env.params.pretty_inspect}\n"
end
```
Trying that cURL request from above, I can see that the server attempted to parse the POST body:
```
#<Kemal::ParamParser:0x2239d70
 @body=HTTP::Params(@raw_params={"{\"testing\":[1,2,3]}" => [""]}),
 @body_parsed=true,
 @files={},
 @json={},
...
```

Is it possible to access the raw POST body as a string?

**Comment by sdogruyol** [comment]
@nick-andren thanks for the issue.

This is kind of an antipattern. Request body is parsed and consumed as `"application/x-www-form-urlencoded"` since the `Content-Type`.

**Comment by nick-andren** [comment]
Hey @sdogruyol, thanks for the quick follow-up. I should have clarified: submitting this with a `Content-Type: applicaiton/json` header is certainly preferred. That said, your comment helped to push me in the right direction since I was not using that.

After some digging, I modified the server code to do the following:
```
post "/testing" do |env|
  puts env.request.body.as(IO).gets_to_end
end
```

I've included `Content-Type` header in the request from above:
```
curl -X POST -d '{"testing":[1,2,3]}' -H "Content-Type: application/json" 0.0.0.0:3000/testing
```

Now I can see the POST body:
```
{"testing":[1,2,3]}
2017-03-04 19:57:46 -0600 200 POST /testing 78.0µs
```

Thanks again for your help!
